sstables/storage.cc

```diff
@@ -139,16 +146,6 @@ void filesystem_storage::open(sstable& sst) {
                                     open_flags::exclusive,
                                     options).get0();
     auto w = file_writer(output_stream<char>(std::move(sink)), std::move(file_path));
-
-    bool toc_exists = file_exists(sst.filename(component_type::TOC)).get0();
```

Comment:
I believe you're breaking the fix to a potential race condition: 0af786f3ea896ef6d61eb4b678800895b6fef0c4

Reply:
ahh, should have thought about this more. as it does look like a transaction-alike thing now. i am dropping this change.